CMakeModules/AddTestHelpers.cmake

```diff
@@ -0,0 +1,39 @@
+function(setup_test_common)
+   cmake_parse_arguments(PARSE_ARGV 0 arg "" "NAME;COST;TIMEOUT" "COMMAND")
+
+   add_test(NAME "${arg_NAME}" COMMAND ${arg_COMMAND} WORKING_DIRECTORY "${CMAKE_BINARY_DIR}")
+
+   if(arg_COST)
+      set_tests_properties("${arg_NAME}" PROPERTIES COST ${arg_COST})
+   endif()
+   if(arg_TIMEOUT)
+      set_tests_properties("${arg_NAME}" PROPERTIES TIMEOUT ${arg_TIMEOUT})
+   endif()
+endfunction()
+
+function(add_p_test)
+   cmake_parse_arguments(PARSE_ARGV 0 arg "" "NAME;COST;TIMEOUT" "COMMAND")
```

Comment:
Is this `cmake_parse_arguments` duplicate as `setup_test_common` calls it too?

Reply:
good eye -- I have to admit I'm not 100% sure on the semantics of `cmake_parse_arguments` so I went with the conservative choice to just repeat all arguments. During development I did test that just `cmake_parse_arguments(PARSE_ARGV 0 arg "" "NAME" "")` worked but I wasn't really sure it worked right in all cases.

That said -- the item you placed the comment on, `add_p_test` does not seem to need to parse anything since unlike `add_wasmspec_test`/`add_np_test`/`add_lr_test` it does not need even `NAME`. So it looks fine to remove this line from `add_p_test`, I will try.